tests/suites/hup/suite.js

```diff
@@ -0,0 +1,346 @@
+  fs.writeFileSync(rollbackLogs, rollback);
+  await that.archiver.add(rollbackLogs);
+
+  const journal = await that.context
+    .get()
+    .worker.executeCommandInHostOS(
+      `journalctl --no-pager --no-hostname --list-boots | awk '{print $1}' | xargs -I{} sh -c 'set -x; journalctl --no-pager --no-hostname -n500 -a -b {};'`,
+      target,
+    );
+  const journalLogs = join(that.suite.options.tmpdir, `journal.log`);
+  fs.writeFileSync(journalLogs, journal);
+  await that.archiver.add(journalLogs);
+}
+
+module.exports = {
+  title: "Hostapp update suite",
+
+  run: async function () {
+    const Worker = this.require("common/worker");
+    const BalenaOS = this.require("components/os/balenaos");
+    const Balena = this.require("components/balena/sdk");
+
+    await fse.ensureDir(this.suite.options.tmpdir);
+
+    this.suite.context.set({
+      utils: this.require("common/utils"),
+      sdk: new Balena(this.suite.options.balena.apiUrl, this.getLogger()),
+      sshKeyPath: join(homedir(), "id"),
+      link: `${this.suite.options.balenaOS.config.uuid.slice(0, 7)}.local`,
+      worker: new Worker(this.suite.deviceType.slug, this.getLogger()),
+    });
+
+    // Network definitions
+    if (this.suite.options.balenaOS.network.wired === true) {
+      this.suite.options.balenaOS.network.wired = {
+        nat: true,
+      };
+    } else {
+      delete this.suite.options.balenaOS.network.wired;
+    }
+    if (this.suite.options.balenaOS.network.wireless === true) {
+      this.suite.options.balenaOS.network.wireless = {
+        ssid: this.suite.options.id,
+        psk: `${this.suite.options.id}_psk`,
+        nat: true,
+      };
+    } else {
+      delete this.suite.options.balenaOS.network.wireless;
+    }
+
+    this.suite.context.set({
+      hup: {
+        doHUP:        doHUP,
+        getOSVersion: getOSVersion,
+        initDUT:      initDUT,
+        runRegistry:  runRegistry,
+        archiveLogs:  archiveLogs,
+      }
+    })
+
+    // Downloads the balenaOS image we hup from
+    const path = await this.context.get().sdk.fetchOS(
+      this.suite.options.balenaOS.download.version,
+      this.suite.deviceType.slug
+    );
+
+    this.suite.context.set({
+      os: new BalenaOS(
+        {
+          deviceType: this.suite.deviceType.slug,
+          network: this.suite.options.balenaOS.network,
+          image: `${path}`,
+          configJson: {
+            uuid: this.suite.options.balenaOS.config.uuid,
+            os: {
+              sshKeys: [
+                await this.context
+                  .get()
+                  .utils.createSSHKey(this.context.get().sshKeyPath),
+              ],
+            },
+            // persistentLogging is managed by the supervisor and only read at first boot
+            persistentLogging: true,
+          },
+        },
+        this.getLogger()
+      ),
+      hupOs: new BalenaOS(
+        {
+          deviceType: this.suite.deviceType.slug,
```

Comment:
this is the `balena-image.docker` file that we're sending from the client isn't it? 

In this case, you can remove all of the fields in the options argument for this constructor, as they aren't going to be used for anything:

```js
hupOs: new BalenaOS(
  {}, // no options needed, this is just a .docker file
  this.getLogger()
),
```

Reply:
Roger that, yeah makes sense.